pkg/pool-stable/test/ComposableStablePool.test.ts

```diff
@@ -686,11 +704,26 @@ describe('ComposableStablePool', () => {
               // Amounts in should be the same as initial ones
               expect(result.amountsIn).to.deep.equal(amountsIn);
 
-              // Make sure received BPT is closed to what we expect
+              // Make sure received BPT is close to what we expect
               const currentBptBalance = await pool.balanceOf(recipient);
               expect(currentBptBalance.sub(previousBptBalance)).to.be.equalWithError(expectedBptOut, 0.0001);
             });
 
+            it('updates the invariant afterward', async () => {
+              const minimumBptOut = pct(expectedBptOut, 0.99);
+
+              const preJoinInvariant = await pool.estimateInvariant();
+              await pool.joinGivenIn({ amountsIn, minimumBptOut, recipient, from: recipient });
+              const expectedLastInvariant = await pool.estimateInvariant();
+              expect(expectedLastInvariant).gt(preJoinInvariant);
```

Comment:
As we're doing a totally proportional join on initialisation we can calculate the invariant exactly as it should be equal to the sum of all the tokens (up to a scaling factor) so we could tighten up the test using that.

Up to you on whether that's worth it.

Reply:
I don't know - people don't necessary have to initialize it proportionally.

If they do initialize disproportionately, we aren't charging them protocol fees (we just update the cache directly). We don't have a proportional join now, so we can't enforce that. Maybe that's another test? That we don't pay protocol fees on disproportionate initialization.